### PR TITLE
Renamed DnsRecord to RecordSet. Fixes #779.

### LINKED ISSUE #779: Rename class DnsRecord
This was pointed out by the DNS team.

### PR DESCRIPTION
**Pull request by mderka**

### CODE REVIEW

File: gcloud-java-dns/README.md [hunk 1]
```diff
@@ -182,12 +182,12 @@ ChangeRequest changeRequest = ChangeRequest.builder().add(toCreate).build();
 changeRequest = zone.applyChangeRequest(changeRequest);
 ```
 
-The `addRecord` method of `DnsRecord.Builder` accepts records in the form of
-strings. The format of the strings depends on the type of the DNS record to be added.
-More information on the supported DNS record types and record formats can be found [here](https://cloud.google.com/dns/what-is-cloud-dns#supported_record_types).
+The `addRecord` method of `RecordSet.Builder` accepts record sets in the form of
```
Comment (mziccard) [inline gcloud-java-dns/README.md]: Here I would say that the method accepts "records", not record sets.

File: gcloud-java-dns/src/main/java/com/google/gcloud/dns/DnsImpl.java [hunk 2]
```diff
@@ -178,27 +178,27 @@ public Zone apply(
   }
 
   @Override
-  public Page<DnsRecord> listDnsRecords(String zoneName, DnsRecordListOption... options) {
+  public Page<RecordSet> listRecordSets(String zoneName, RecordSetListOption... options) {
     return listDnsRecords(zoneName, options(), optionMap(options));
   }
 
-  private static Page<DnsRecord> listDnsRecords(final String zoneName,
+  private static Page<RecordSet> listDnsRecords(final String zoneName,
```
Comment (mziccard) [inline gcloud-java-dns/src/main/java/com/google/gcloud/dns/DnsImpl.java]: s/listDnsRecords/listRecordSets

File: gcloud-java-dns/src/main/java/com/google/gcloud/dns/DnsImpl.java [hunk 3]
```diff
@@ -178,27 +178,27 @@ public Zone apply(
...
+              return rpc.listRecordSets(zoneName, optionsMap);
             }
           }, serviceOptions.retryParams(), EXCEPTION_HANDLER);
       String cursor = result.pageToken();
-      // transform that list into dns records
-      Iterable<DnsRecord> records = result.results() == null
-          ? ImmutableList.<DnsRecord>of()
-          : Iterables.transform(result.results(), DnsRecord.FROM_PB_FUNCTION);
+      // transform that list into record sets
+      Iterable<RecordSet> records = result.results() == null
```
Comment (mziccard) [inline gcloud-java-dns/src/main/java/com/google/gcloud/dns/DnsImpl.java]: s/records/recordSets

File: gcloud-java-dns/src/main/java/com/google/gcloud/dns/RecordSet.java [hunk 4]
```diff
@@ -35,27 +34,29 @@
...
  *
  * @see <a href="https://cloud.google.com/dns/api/v1/resourceRecordSets">Google Cloud DNS
  * documentation</a>
  */
-public class DnsRecord implements Serializable {
+public class RecordSet implements Serializable {
 
-  static final Function<ResourceRecordSet, DnsRecord> FROM_PB_FUNCTION =
-      new Function<com.google.api.services.dns.model.ResourceRecordSet, DnsRecord>() {
+  static final Function<com.google.api.services.dns.model.ResourceRecordSet, RecordSet>
```
Comment (mziccard) [inline gcloud-java-dns/src/main/java/com/google/gcloud/dns/RecordSet.java]: You can import `ResourceRecordSet` and make this and other lines shorter

File: gcloud-java-dns/src/main/java/com/google/gcloud/dns/RecordSet.java [hunk 5]
```diff
@@ -35,27 +34,29 @@
...
         }
       };
-  static final Function<DnsRecord, ResourceRecordSet> TO_PB_FUNCTION =
-      new Function<DnsRecord, ResourceRecordSet>() {
+  static final Function<RecordSet, com.google.api.services.dns.model.ResourceRecordSet>
+      TO_PB_FUNCTION =
+      new Function<RecordSet, com.google.api.services.dns.model.ResourceRecordSet>() {
         @Override
-        public com.google.api.services.dns.model.ResourceRecordSet apply(DnsRecord error) {
+        public com.google.api.services.dns.model.ResourceRecordSet apply(RecordSet error) {
```
Comment (mziccard) [inline gcloud-java-dns/src/main/java/com/google/gcloud/dns/RecordSet.java]: Not related to this PR, why is the variable called `error`?

File: gcloud-java-dns/src/main/java/com/google/gcloud/dns/RecordSet.java [hunk 6]
```diff
@@ -186,7 +187,7 @@ public Builder records(List<String> records) {
     }
 
     /**
-     * Sets name for this DNS record set. For example, www.example.com.
+     * Sets name for this record set. For example, www.example.com.
```
Comment (mziccard) [inline gcloud-java-dns/src/main/java/com/google/gcloud/dns/RecordSet.java]: Nit: "Sets name" -> "Sets the name"

File: gcloud-java-dns/src/main/java/com/google/gcloud/dns/RecordSet.java [hunk 7]
```diff
@@ -282,7 +283,8 @@ public int hashCode() {
 
   @Override
   public boolean equals(Object obj) {
-    return (obj instanceof DnsRecord) && Objects.equals(this.toPb(), ((DnsRecord) obj).toPb());
+    return obj instanceof RecordSet
+        && Objects.equals(this.toPb(), ((RecordSet) obj).toPb());
```
Comment (mziccard) [inline gcloud-java-dns/src/main/java/com/google/gcloud/dns/RecordSet.java]: Should fit on the same line.

File: gcloud-java-dns/src/test/java/com/google/gcloud/dns/DnsTest.java [hunk 8]
```diff
@@ -33,33 +33,33 @@
   public void testDnsRecordListOption() {
     // dns name
     String dnsName = "some name";
-    Dns.DnsRecordListOption dnsRecordListOption = Dns.DnsRecordListOption.dnsName(dnsName);
-    assertEquals(dnsName, dnsRecordListOption.value());
-    assertEquals(DnsRpc.Option.NAME, dnsRecordListOption.rpcOption());
+    Dns.RecordSetListOption rrsetListOption = Dns.RecordSetListOption.dnsName(dnsName);
```
Comment (mziccard) [inline gcloud-java-dns/src/test/java/com/google/gcloud/dns/DnsTest.java]: Nit: my preference would be for calling this `recordSetListOption` (which is no longer than the previous `dnsRecordListOption`). I feel like `rrset` is not very human-readable.

File: gcloud-java-dns/src/test/java/com/google/gcloud/dns/RecordSetTest.java [hunk 9]
```diff
@@ -26,20 +26,20 @@
...
-  private static final DnsRecord record = builder(NAME, TYPE)
+  private static final RecordSet.Type TYPE = RecordSet.Type.AAAA;
+  private static final RecordSet recordSet = builder(NAME, TYPE)
       .ttl(UNIT_TTL, UNIT)
       .build();
 
   @Test
   public void testDefaultDnsRecord() {
-    DnsRecord record = builder(NAME, TYPE).build();
+    RecordSet record = builder(NAME, TYPE).build();
```
Comment (mziccard) [inline gcloud-java-dns/src/test/java/com/google/gcloud/dns/RecordSetTest.java]: Nit (only if you have time): here and in the following of this class variable names should be like `recordSet`, not `record`.

File: gcloud-java-dns/src/test/java/com/google/gcloud/dns/it/ITDnsTest.java [hunk 10]
```diff
@@ -98,11 +98,12 @@ private static void clear() {
         while (iterator.hasNext()) {
           waitForChangeToComplete(zoneName, iterator.next().id());
         }
-        Iterator<DnsRecord> recordIterator = zone.listDnsRecords().iterateAll();
-        List<DnsRecord> toDelete = new LinkedList<>();
+        Iterator<RecordSet> recordIterator = zone.listRecordSets().iterateAll();
```
Comment (mziccard) [inline gcloud-java-dns/src/test/java/com/google/gcloud/dns/it/ITDnsTest.java]: Nit `recordSetIterator` (also elsewhere in this class)

File: gcloud-java-dns/src/test/java/com/google/gcloud/dns/it/ITDnsTest.java [hunk 11]
```diff
@@ -98,11 +98,12 @@ private static void clear() {
...
+        List<RecordSet> toDelete = new LinkedList<>();
         while (recordIterator.hasNext()) {
-          DnsRecord record = recordIterator.next();
-          if (!ImmutableList.of(DnsRecord.Type.NS, DnsRecord.Type.SOA).contains(record.type())) {
+          RecordSet record = recordIterator.next();
```
Comment (mziccard) [inline gcloud-java-dns/src/test/java/com/google/gcloud/dns/it/ITDnsTest.java]: Nit `recordSet` (also elsewhere in this class)

File: gcloud-java-dns/src/test/java/com/google/gcloud/dns/it/ITDnsTest.java [hunk 12]
```diff
@@ -831,77 +833,78 @@ public void testGetProject() {
...
+      assertEquals(2, recordSets.size());
+      ImmutableList<RecordSet.Type> defaultRecords =
+          ImmutableList.of(RecordSet.Type.NS, RecordSet.Type.SOA);
+      for (RecordSet record : recordSets) {
         assertTrue(defaultRecords.contains(record.type()));
       }
       // field options
-      Iterator<DnsRecord> dnsRecordIterator = DNS.listDnsRecords(zone.name(),
-          Dns.DnsRecordListOption.fields(Dns.DnsRecordField.TTL)).iterateAll();
+      Iterator<RecordSet> dnsRecordIterator = DNS.listRecordSets(zone.name(),
```
Comment (mziccard) [inline gcloud-java-dns/src/test/java/com/google/gcloud/dns/it/ITDnsTest.java]: Nit `recordSetIterator`  (also elsewhere in this class)

File: gcloud-java-dns/src/test/java/com/google/gcloud/dns/it/ITDnsTest.java [hunk 13]
```diff
@@ -831,77 +833,78 @@ public void testGetProject() {
...
         assertTrue(record.records().isEmpty());
         assertNull(record.ttl());
         counter++;
       }
       assertEquals(2, counter);
       // test page size
-      Page<DnsRecord> dnsRecordPage = DNS.listDnsRecords(zone.name(),
-          Dns.DnsRecordListOption.fields(Dns.DnsRecordField.TYPE),
-          Dns.DnsRecordListOption.pageSize(1));
+      Page<RecordSet> dnsRecordPage = DNS.listRecordSets(zone.name(),
```
Comment (mziccard) [inline gcloud-java-dns/src/test/java/com/google/gcloud/dns/it/ITDnsTest.java]: Nit `recordSetPage`

File: gcloud-java-examples/src/main/java/com/google/gcloud/examples/dns/DnsExample.java [hunk 14]
```diff
@@ -203,7 +203,7 @@ public void run(Dns dns, String... args) {
       if (args.length > 3) {
         ttl = Integer.parseInt(args[3]);
       }
-      DnsRecord record = DnsRecord.builder(recordName, DnsRecord.Type.A)
+      RecordSet record = RecordSet.builder(recordName, RecordSet.Type.A)
```
Comment (mziccard) [inline gcloud-java-examples/src/main/java/com/google/gcloud/examples/dns/DnsExample.java]: Nit `recordSet`

File: gcloud-java-examples/src/main/java/com/google/gcloud/examples/dns/snippets/CreateOrUpdateRecordSets.java [hunk 15]
```diff
@@ -59,9 +59,9 @@ public static void main(String... args) {
 
     // Verify a www.<zone-domain>. type A record does not exist yet.
     // If it does exist, we will overwrite it with our prepared record.
-    Iterator<DnsRecord> recordIterator = zone.listDnsRecords().iterateAll();
+    Iterator<RecordSet> recordIterator = zone.listRecordSets().iterateAll();
```
Comment (mziccard) [inline gcloud-java-examples/src/main/java/com/google/gcloud/examples/dns/snippets/CreateOrUpdateRecordSets.java]: Nit `recordSetIterator`

**Comment by mziccard** [comment]
You might already have noticed but CI is failing:

```
[ERROR] /home/travis/build/GoogleCloudPlatform/gcloud-java/gcloud-java-dns/src/test/java/com/google/gcloud/dns/SerializationTest.java:[94,42] cannot find symbol
  symbol:   variable DNS_RECORD_PARTIAL
  location: class com.google.gcloud.dns.SerializationTest
[ERROR] /home/travis/build/GoogleCloudPlatform/gcloud-java/gcloud-java-dns/src/test/java/com/google/gcloud/dns/SerializationTest.java:[94,62] cannot find symbol
  symbol:   variable DNS_RECORD_COMPLETE
  location: class com.google.gcloud.dns.SerializationTest
```

File: gcloud-java-dns/README.md (hunk 1 above)
Comment (mderka) [inline gcloud-java-dns/README.md]: Done.

File: gcloud-java-dns/src/main/java/com/google/gcloud/dns/DnsImpl.java (hunk 2 above)
Comment (mderka) [inline gcloud-java-dns/src/main/java/com/google/gcloud/dns/DnsImpl.java]: Done.

File: gcloud-java-dns/src/main/java/com/google/gcloud/dns/DnsImpl.java (hunk 3 above)
Comment (mderka) [inline gcloud-java-dns/src/main/java/com/google/gcloud/dns/DnsImpl.java]: Done.

File: gcloud-java-dns/src/main/java/com/google/gcloud/dns/RecordSet.java (hunk 4 above)
Comment (mderka) [inline gcloud-java-dns/src/main/java/com/google/gcloud/dns/RecordSet.java]: Done.

File: gcloud-java-dns/src/main/java/com/google/gcloud/dns/RecordSet.java (hunk 5 above)
Comment (mderka) [inline gcloud-java-dns/src/main/java/com/google/gcloud/dns/RecordSet.java]: No idea how it got there. Fixed.

File: gcloud-java-dns/src/main/java/com/google/gcloud/dns/RecordSet.java (hunk 6 above)
Comment (mderka) [inline gcloud-java-dns/src/main/java/com/google/gcloud/dns/RecordSet.java]: done

File: gcloud-java-dns/src/main/java/com/google/gcloud/dns/RecordSet.java (hunk 7 above)
Comment (mderka) [inline gcloud-java-dns/src/main/java/com/google/gcloud/dns/RecordSet.java]: Done.

File: gcloud-java-dns/src/test/java/com/google/gcloud/dns/DnsTest.java (hunk 8 above)
Comment (mderka) [inline gcloud-java-dns/src/test/java/com/google/gcloud/dns/DnsTest.java]: Done. Also renamed the test method.

File: gcloud-java-dns/src/test/java/com/google/gcloud/dns/it/ITDnsTest.java (hunk 10 above)
Comment (mderka) [inline gcloud-java-dns/src/test/java/com/google/gcloud/dns/it/ITDnsTest.java]: Fixed.

File: gcloud-java-dns/src/test/java/com/google/gcloud/dns/RecordSetTest.java (hunk 9 above)
Comment (mderka) [inline gcloud-java-dns/src/test/java/com/google/gcloud/dns/RecordSetTest.java]: Fixed.

File: gcloud-java-dns/src/test/java/com/google/gcloud/dns/it/ITDnsTest.java (hunk 11 above)
Comment (mderka) [inline gcloud-java-dns/src/test/java/com/google/gcloud/dns/it/ITDnsTest.java]: Done.

File: gcloud-java-dns/src/test/java/com/google/gcloud/dns/it/ITDnsTest.java (hunk 12 above)
Comment (mderka) [inline gcloud-java-dns/src/test/java/com/google/gcloud/dns/it/ITDnsTest.java]: Done.

File: gcloud-java-dns/src/test/java/com/google/gcloud/dns/it/ITDnsTest.java (hunk 13 above)
Comment (mderka) [inline gcloud-java-dns/src/test/java/com/google/gcloud/dns/it/ITDnsTest.java]: Done.

File: gcloud-java-examples/src/main/java/com/google/gcloud/examples/dns/DnsExample.java (hunk 14 above)
Comment (mderka) [inline gcloud-java-examples/src/main/java/com/google/gcloud/examples/dns/DnsExample.java]: Fixed.

File: gcloud-java-examples/src/main/java/com/google/gcloud/examples/dns/snippets/CreateOrUpdateRecordSets.java (hunk 15 above)
Comment (mderka) [inline gcloud-java-examples/src/main/java/com/google/gcloud/examples/dns/snippets/CreateOrUpdateRecordSets.java]: Done.

**Comment by mziccard** [comment]
I finished my pass

**Comment by mderka** [comment]
Done with revisions. Ready to proceed.

File: gcloud-java-dns/src/test/java/com/google/gcloud/dns/it/ITDnsTest.java [hunk 16]
```diff
@@ -608,22 +608,22 @@ public void testInvalidChangeRequest() {
...
-        fail("Deleted a record without a complete match.");
+        fail("Deleted a record set without a complete match.");
       } catch (DnsException ex) {
         // expected
         assertEquals(412, ex.code());
         assertFalse(ex.retryable());
       }
       // delete and add SOA
-      Iterator<RecordSet> recordIterator = zone.listRecordSets().iterateAll();
+      Iterator<RecordSet> recordSetIteratorrator = zone.listRecordSets().iterateAll();
```
Comment (mziccard) [inline gcloud-java-dns/src/test/java/com/google/gcloud/dns/it/ITDnsTest.java]: `Iteratorrator` is an amazing name but we better go with `Iterator` :)

**Comment by mziccard** [comment]
Added one more nit, feel free to merge when done!

File: gcloud-java-dns/src/test/java/com/google/gcloud/dns/it/ITDnsTest.java (hunk 16 above)
Comment (mderka) [inline gcloud-java-dns/src/test/java/com/google/gcloud/dns/it/ITDnsTest.java]: :-( Fixed.

**Comment by mderka** [comment]
Fixed and squashed commits. I will merge when travis finished. Thanks.

**Pull request merged**